packages/nimble-components/src/rich-text-mention/users/tests/rich-text-mention-users.spec.ts

```diff
@@ -0,0 +1,348 @@
+            expect(element.validity.missingDisplayNameValue).toBeTrue();
+            expect(element.mentionInternals.mentionConfig).toBe(undefined);
+        });
+
+        it('is invalid with duplicate key values', async () => {
+            ({ element, connect, disconnect } = await setup([
+                { key: 'user:1', displayName: 'user' },
+                { key: 'user:1', displayName: 'user' }
+            ]));
+            await connect();
+            expect(element.checkValidity()).toBeFalse();
+            expect(element.validity.duplicateMappingMentionHref).toBeTrue();
+            expect(element.mentionInternals.mentionConfig).toBe(undefined);
+        });
+
+        it('is invalid with mismatch pattern', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'user' },
+                    { key: 'user:2', displayName: 'user' }
+                ],
+                'invalidPattern'
+            ));
+            await connect();
+            expect(element.checkValidity()).toBeFalse();
+            expect(element.validity.unsupportedMentionHrefValue).toBeTrue();
+            expect(element.mentionInternals.mentionConfig).toBe(undefined);
+        });
+
+        it('is invalid with invalid mismatching key', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'invalid_userkey', displayName: 'user' },
+                    { key: 'user:2', displayName: 'user' }
+                ],
+                'user:'
+            ));
+            await connect();
+            expect(element.checkValidity()).toBeFalse();
+            expect(element.validity.unsupportedMentionHrefValue).toBeTrue();
+            expect(element.mentionInternals.mentionConfig).toBe(undefined);
+        });
+
+        it('is valid with valid pattern', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'user' },
+                    { key: 'user:2', displayName: 'user' }
+                ],
+                'user:.*'
+            ));
+            await connect();
+            expect(element.checkValidity()).toBeTrue();
+            expect(element.validity.unsupportedMentionHrefValue).toBeFalse();
+        });
+
+        it('is invalid with invalid regex pattern', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'user' },
+                    { key: 'user:2', displayName: 'user' }
+                ],
+                '(invalid'
+            ));
+            await connect();
+            expect(element.checkValidity()).toBeFalse();
+            expect(element.validity.unsupportedPatternValue).toBeTrue();
+            expect(element.validity.unsupportedMentionHrefValue).toBeTrue();
```

Comment:
If it's not too much trouble, I think we should not be setting `unsupportedMentionHrefValue` when we know the pattern is invalid.

Reply:
Updated it to avoid setting the `unsupportedMentionHrefValue` when the pattern is invalid.